core/mutator/entry/mutator.go

```diff
@@ -88,7 +89,8 @@ func ReduceFn(leaves []*pb.EntryUpdate, msgs []*pb.EntryUpdate,
 	for i, msg := range msgs {
 		newValue, err := MutateFn(oldValue, msg.GetMutation())
 		if err != nil {
-			emitErr(fmt.Errorf("entry: ReduceFn(msg %d/%d): %v", i, len(msgs)-1, err))
+			s := status.Convert(err)
+			emitErr(status.Errorf(s.Code(), "entry: ReduceFn(msg %d/%d): %v", i, len(msgs)-1, s.Message()))
```

Comment:
Should it be `len(msgs)` without `-1`?

Reply:
i iterates from 0 to n-1, so this either needs to be n-1 or i+1.  Perhaps +1 is better.

Reply:
Hm, don't know. Still confusing a bit, but I think any syntax of this kind if ambiguous. Leaving this up to you.